mypy/semanal.py

```diff
@@ -6408,14 +6408,44 @@ def lookup_fully_qualified_or_none(self, fullname: str) -> SymbolTableNode | Non
         # TODO: support nested classes (but consider performance impact,
         #       we might keep the module level only lookup for thing like 'builtins.int').
         assert "." in fullname
-        module, name = fullname.rsplit(".", maxsplit=1)
-        if module not in self.modules:
+
+        splitted_modules = fullname.rsplit(".")
+        names = []
+
+        while splitted_modules and ".".join(splitted_modules) not in self.modules:
+            """
+            Try to find the module in the modules dictionary.
+
+            If the module is not found, pop the last element of the splitted list and append it to the names list.
+
+            This is to find the longest prefix of the module name that is in the modules dictionary.
+            """
```

Comment:
Style nit: We only use docstrings at beginning of a function/module/class, not inside function bodies.